### PR TITLE
add apim policy, portal files and scripts

### PR DESCRIPTION
**Pull request by l-pinna**

### CODE REVIEW

File: azurerm_api_management/output.tf [hunk 1]
```diff
@@ -1,4 +1,5 @@
-output "id" {
-  description = "The Azure resource group id"
-  value       = "${module.azurerm_api_management.id}"
-}
+# output "id" {
```
Comment (LucaPrete) [inline azurerm_api_management/output.tf]: please, remove commented output if not needed. Put a comment instead...or add relevant output (maybe at least the name of the api manager created?)

File: azurerm_api_management/variables.tf [hunk 2]
```diff
@@ -15,36 +10,25 @@ variable "resource_name_prefix" {
...
+  description = "The API Management name."
 }
 
-# TF_VAR_ADB2C_TENANT_ID
-variable "ADB2C_TENANT_ID" {
-  type        = "string"
-  description = "Name of the Active Directory B2C tenant used in the API management portal authentication flow"
-  default     = "agidweb"
-}
+locals {
```
Comment (LucaPrete) [inline azurerm_api_management/variables.tf]: first the variables. Then the locals, at the bottom

File: azurerm_api_management_apis/main.tf [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+# Existing infrastructure
+data "azurerm_resource_group" "rg" {
+  name = "${local.azurerm_resource_group_name}"
+}
+
+data "azurerm_api_management" "api_management" {
+  name                = "${local.azurerm_apim_name}"
+  resource_group_name = "${data.azurerm_resource_group.rg.name}"
+}
+
```
Comment (LucaPrete) [inline azurerm_api_management_apis/main.tf]: Add white line, then # New infrastructure, then white line

File: azurerm_api_management_apis/main.tf [hunk 4]
```diff
@@ -0,0 +1,60 @@
+# API management
+## Create and configure the API management service
+# Existing infrastructure
```
Comment (LucaPrete) [inline azurerm_api_management_apis/main.tf]: Add a white line before and after

File: azurerm_api_management_apis/main.tf [hunk 5]
```diff
@@ -0,0 +1,60 @@
+# API management
+## Create and configure the API management service
```
Comment (LucaPrete) [inline azurerm_api_management_apis/main.tf]: Keep only one #

File: azurerm_api_management_apis/main.tf [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  api_management_name = "${basename(data.azurerm_api_management.api_management.id)}"
+  resource_group_name = "${data.azurerm_resource_group.rg.name}"
+
+  display_name = "${lookup(var.apim_apis[count.index],"display_name","${lookup(var.apim_apis[count.index],"name")}")}"
+  description  = "${lookup(var.apim_apis[count.index],"description","---")}"
+  revision     = "${lookup(var.apim_apis[count.index],"revision","1")}"
+  path         = "${lookup(var.apim_apis[count.index],"path","api/v1")}"
+  protocols    = ["${split(",",lookup(var.apim_apis[count.index],"protocols","https"))}"]
+
+  # import {
```
Comment (LucaPrete) [inline azurerm_api_management_apis/main.tf]: Remove commented code

File: azurerm_api_management_apis/main.tf [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+  #   content_value  = "http://conferenceapi.azurewebsites.net/?format=json"
+  # }
+}
+
+resource "azurerm_api_management_api_operation" "apim_api_operations" {
+  count        = "${length(var.apim_api_operations)}"
+  operation_id = "${lookup(var.apim_api_operations[count.index],"operation_id")}"
+  api_name     = "${element(azurerm_api_management_api.apim_apis.*.name , index(azurerm_api_management_api.apim_apis.*.name, lookup(var.apim_api_operations[count.index],"api_name")))}"
+
+  # api_name            = "${lookup(var.apim_api_operations[count.index],"api_name")}"
```
Comment (LucaPrete) [inline azurerm_api_management_apis/main.tf]: Remove commented code

File: azurerm_api_management_apis/main.tf [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  api_management_name = "${basename(data.azurerm_api_management.api_management.id)}"
+  resource_group_name = "${data.azurerm_resource_group.rg.name}"
+  display_name        = "${lookup(var.apim_api_operations[count.index],"display_name","${lookup(var.apim_api_operations[count.index],"api_name")}")}"
+  method              = "${lookup(var.apim_api_operations[count.index],"method","GET")}"
+  url_template        = "${lookup(var.apim_api_operations[count.index],"url_template")}"
+  description         = "${lookup(var.apim_apis[count.index],"description","---")}"
+
+  # response           = {}
```
Comment (LucaPrete) [inline azurerm_api_management_apis/main.tf]: remove all commented code

File: azurerm_api_management_apis/main.tf [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+## 27868
+resource "azurerm_api_management_api_operation_policy" "apim_api_operation_policies" {
+  count               = "${length(var.apim_api_operations)}"
+  api_name            = "${lookup(var.apim_api_operations[count.index],"api_name")}"
+  api_management_name = "${basename(data.azurerm_api_management.api_management.id)}"
+  resource_group_name = "${data.azurerm_resource_group.rg.name}"
+
+  # operation_id        = "${lookup(var.apim_api_operations[count.index],"operation_id")}"
```
Comment (LucaPrete) [inline azurerm_api_management_apis/main.tf]: remove commented code

File: azurerm_api_management_apis/main.tf [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+  method              = "${lookup(var.apim_api_operations[count.index],"method","GET")}"
+  url_template        = "${lookup(var.apim_api_operations[count.index],"url_template")}"
+  description         = "${lookup(var.apim_apis[count.index],"description","---")}"
+
+  # response           = {}
+  # request            = {}
+  # template_parameter = {}
+}
+
+## 27868
```
Comment (LucaPrete) [inline azurerm_api_management_apis/main.tf]: remove commented code

File: azurerm_api_management_apis/output.tf [hunk 11]
```diff
@@ -0,0 +1,5 @@
+# output "id" {
```
Comment (LucaPrete) [inline azurerm_api_management_apis/output.tf]: Same comment of the other output file

File: azurerm_api_management_apis/tg.tf [hunk 12]
```diff
@@ -0,0 +1,11 @@
+provider "azurerm" {
+  version = "~>1.29"
+}
+
+provider "azuread" {
+  version = "~>0.2"
```
Comment (LucaPrete) [inline azurerm_api_management_apis/tg.tf]: we can increment to 1.32.1 and 0.5

File: azurerm_api_management_apis/variables.tf [hunk 13]
```diff
@@ -0,0 +1,31 @@
...
+
+variable "resource_name_prefix" {
+  description = "The prefix used to name all resources created."
+}
+
+variable "apim_name" {
+  description = "The API Management name."
+}
+
+locals {
```
Comment (LucaPrete) [inline azurerm_api_management_apis/variables.tf]: locals at the bottom

File: azurerm_api_management_apis/variables.tf [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+  # Define resource names based on the following convention:  # {azurerm_resource_name_prefix}-RESOURCE_TYPE-{environment}
+  azurerm_resource_group_name = "${var.resource_name_prefix}-${var.environment}-rg"
+  azurerm_apim_name           = "${var.resource_name_prefix}-${var.environment}-apim-${var.apim_name}"
+}
+
+variable "apim_apis" {
+  description = "Apis."
```
Comment (LucaPrete) [inline azurerm_api_management_apis/variables.tf]: APIs

File: azurerm_api_management_products/main.tf [hunk 15]
```diff
@@ -0,0 +1,45 @@
+# API management
```
Comment (LucaPrete) [inline azurerm_api_management_products/main.tf]: Same comment of the previous main.tf header

File: azurerm_api_management_products/main.tf [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+## Create and configure the API management service
```
Comment (LucaPrete) [inline azurerm_api_management_products/main.tf]: remove all double #. Keep only one

File: azurerm_api_management_products/main.tf [hunk 17]
```diff
@@ -0,0 +1,45 @@
...
+  subscription_required = "${lookup(var.apim_products[count.index],"subscription_required","true")}"
+  subscriptions_limit   = "${lookup(var.apim_products[count.index],"subscriptions_limit","100")}"
+  approval_required     = "${lookup(var.apim_products[count.index],"approval_required","true")}"
+  published             = "${lookup(var.apim_products[count.index],"published","true")}"
+}
+
+resource "azurerm_api_management_product_policy" "product_policies" {
+  count = "${length(var.apim_products)}"
+
+  # product_id          = "${basename(element(azurerm_api_management_product.products.*.id,count.index))}"
```
Comment (LucaPrete) [inline azurerm_api_management_products/main.tf]: Remove commented code

File: azurerm_api_management_products/output.tf [hunk 18]
```diff
@@ -0,0 +1,5 @@
+# output "id" {
```
Comment (LucaPrete) [inline azurerm_api_management_products/output.tf]: Same comment of other outputs

File: azurerm_api_management_products/tg.tf [hunk 19]
```diff
@@ -0,0 +1,11 @@
+provider "azurerm" {
+  version = "~>1.29"
```
Comment (LucaPrete) [inline azurerm_api_management_products/tg.tf]: upgrade to 1.32.1 and 0.5

File: azurerm_api_management_products/variables.tf [hunk 20]
```diff
@@ -0,0 +1,32 @@
+variable "environment" {
+  description = "The nick name identifying the type of environment (i.e. test, staging, production)"
```
Comment (LucaPrete) [inline azurerm_api_management_products/variables.tf]: . at the end

File: azurerm_api_management_products/variables.tf [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+
+variable "resource_name_prefix" {
+  description = "The prefix used to name all resources created."
+}
+
+variable "apim_name" {
+  description = "The API Management name."
+}
+
+locals {
```
Comment (LucaPrete) [inline azurerm_api_management_products/variables.tf]: locals at the end

File: azurerm_api_management_products/variables.tf [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+  # Define resource names based on the following convention:  # {azurerm_resource_name_prefix}-RESOURCE_TYPE-{environment}
```
Comment (LucaPrete) [inline azurerm_api_management_products/variables.tf]: Maybe this should be like other files on two lines?

File: azurerm_api_management_properties/main.tf [hunk 23]
```diff
@@ -0,0 +1,36 @@
+# API management
```
Comment (LucaPrete) [inline azurerm_api_management_properties/main.tf]: Same as other main.tf headers

File: azurerm_api_management_properties/output.tf [hunk 24]
```diff
@@ -0,0 +1,5 @@
+# output "id" {
```
Comment (LucaPrete) [inline azurerm_api_management_properties/output.tf]: Same as other outputs

File: azurerm_api_management_properties/tg.tf [hunk 25]
```diff
@@ -0,0 +1,11 @@
+provider "azurerm" {
```
Comment (LucaPrete) [inline azurerm_api_management_properties/tg.tf]: same as other tg.tf

File: azurerm_api_management_properties/variables.tf [hunk 26]
```diff
@@ -0,0 +1,58 @@
...
+
+variable "resource_name_prefix" {
+  description = "The prefix used to name all resources created."
+}
+
+variable "apim_name" {
+  description = "The API Management name."
+}
+
+locals {
```
Comment (LucaPrete) [inline azurerm_api_management_properties/variables.tf]: same as other variables.tf :)

File: azurerm_function_app_config/main.tf [hunk 27]
```diff
@@ -63,27 +63,27 @@ module "azurerm_function_app_VirtualNetwork" {
   }
 }
 
-module "azurerm_function_app_VirtualNetworkConnections" {
-  source      = "git@github.com:teamdigitale/terraform-azurerm-resource.git"
-  api_version = "2015-08-01"
-  type        = "Microsoft.Web/sites/virtualNetworkConnections"
+# module "azurerm_function_app_VirtualNetworkConnections" {
```
Comment (LucaPrete) [inline azurerm_function_app_config/main.tf]: Please, do not commit commented code..

File: azurerm_api_management/output.tf [hunk 28]
```diff
@@ -1,4 +1,4 @@
-output "id" {
-  description = "The Azure resource group id"
-  value       = "${module.azurerm_api_management.id}"
+output "name" {
```
Comment (LucaPrete) [inline azurerm_api_management/output.tf]: Please use azurerm_api_management instead

File: azurerm_api_management/tg.tf [hunk 29]
```diff
@@ -2,10 +2,6 @@ provider "azurerm" {
   version = "~>1.29"
```
Comment (LucaPrete) [inline azurerm_api_management/tg.tf]: 1.32.1

File: azurerm_api_management_apis/main.tf (hunk 8 above)
Comment (LucaPrete) [reply azurerm_api_management_apis/main.tf]: Still needs to be removed

File: azurerm_api_management_apis/tg.tf [hunk 30]
```diff
@@ -0,0 +1,7 @@
+provider "azurerm" {
+  version = "~>1.29"
```
Comment (LucaPrete) [inline azurerm_api_management_apis/tg.tf]: 1.32.1

File: azurerm_api_management_apis/variables.tf (hunk 14 above)
Comment (LucaPrete) [reply azurerm_api_management_apis/variables.tf]: Still needs to be capitalized

File: azurerm_api_management_products/main.tf (hunk 17 above)
Comment (LucaPrete) [reply azurerm_api_management_products/main.tf]: Still to be removed

File: azurerm_api_management_products/output.tf (hunk 18 above)
Comment (LucaPrete) [reply azurerm_api_management_products/output.tf]: Still needs to be fixed

File: azurerm_api_management_products/tg.tf (hunk 19 above)
Comment (LucaPrete) [reply azurerm_api_management_products/tg.tf]: Still needs to be updated

File: azurerm_api_management_products/variables.tf (hunk 20 above)
Comment (LucaPrete) [reply azurerm_api_management_products/variables.tf]: Still needs to be fixed

File: azurerm_api_management_properties/output.tf (hunk 24 above)
Comment (LucaPrete) [reply azurerm_api_management_properties/output.tf]: Still needs to be changed

File: azurerm_api_management_properties/tg.tf (hunk 25 above)
Comment (LucaPrete) [reply azurerm_api_management_properties/tg.tf]: Still needs to be updated

File: azurerm_function_app_config/main.tf (hunk 27 above)
Comment (LucaPrete) [reply azurerm_function_app_config/main.tf]: Still needs to be fixed

File: azurerm_api_management_products/variables.tf (hunk 20 above)
Comment (l-pinna) [reply azurerm_api_management_products/variables.tf]: has been copied from other modules, need a full replace

File: azurerm_api_management/output.tf (hunk 28 above)
Comment (l-pinna) [reply azurerm_api_management/output.tf]: the output from the arm template cannot be retrieved

**Pull request merged**